overlord/hookstate/ctlcmd/get.go

```diff
@@ -22,23 +22,32 @@ package ctlcmd
 import (
 	"encoding/json"
 	"fmt"
+	"strings"
 
 	"github.com/snapcore/snapd/i18n/dumb"
 	"github.com/snapcore/snapd/overlord/configstate"
+	"github.com/snapcore/snapd/overlord/hookstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
 )
 
 type getCommand struct {
 	baseCommand
 
+	// these two options are mutually exclusive
+	ForceSlotSide bool `long:"slot" description:"request attribute of the slot"`
+	ForcePlugSide bool `long:"plug" description:"request attribute of the plug"`
+
 	Positional struct {
-		Keys []string `positional-arg-name:"<keys>" description:"option keys" required:"yes"`
-	} `positional-args:"yes" required:"yes"`
+		PlugOrSlotSpec string   `positional-args:"true" positional-arg-name:":<plug|slot>" required:"yes"`
```

Comment:
Not sure if it's a good idea to have "required" here, as it may produce an awkward error message saying that "<plug|slot> is required", which isn't actually true. Might be better to handle that in code inside Execute.

Might be good to give it a try for real to see how the tool reports help and errors.